autodispose/src/main/java/com/uber/autodispose/AutoDispose.java

```diff
@@ -166,7 +166,7 @@
     return new LifecycleScopeProviderHandlerImpl(scope);
   }
 
-  private static class MaybeScopeHandlerImpl implements ScopeHandler {
+  private static final class MaybeScopeHandlerImpl implements ScopeHandler {
```

Comment:
I'd avoid `private` in case you don't want to get accessor methods on Android.

Reply:
Good catch. Fixed in 959fe09 but planning to remove this class in the near future anyway now that `as()` is in RxJava :)